mm2src/coins/eth/eth_balance_events.rs

```diff
@@ -16,49 +17,79 @@ use super::EthCoin;
 use crate::{eth::{u256_to_big_decimal, Erc20TokenInfo},
             BalanceError, MmCoin};
 
+type BalanceResult = Result<(String, HashMap<String, BigDecimal>), (String, MmError<BalanceError>)>;
+
 /// This implementation differs from others, as they immediately return
 /// an error if any of the requests fails. This one completes all futures
 /// and returns their results individually.
-async fn get_all_balance_results_concurrently(
-    coin: &EthCoin,
-) -> Vec<Result<(String, BigDecimal), (String, MmError<BalanceError>)>> {
+async fn get_all_balance_results_concurrently(coin: &EthCoin) -> Vec<BalanceResult> {
     let mut tokens = coin.get_erc_tokens_infos();
 
-    // Workaround for performance purposes.
-    //
-    // Unlike tokens, the platform coin length is constant (=1). Instead of creating a generic
-    // type and mapping the platform coin and the entire token list (which can grow at any time), we map
-    // the platform coin to Erc20TokenInfo so that we can use the token list right away without
-    // additional mapping.
-    tokens.insert(coin.ticker.clone(), Erc20TokenInfo {
-        token_address: coin.my_address,
-        decimals: coin.decimals,
-    });
-
-    let jobs = tokens
-        .into_iter()
-        .map(|(token_ticker, info)| async move { fetch_balance(coin, token_ticker, &info).await })
-        .collect::<FuturesUnordered<_>>();
-
-    jobs.collect().await
+    let addresses = match coin.my_addresses().await {
+        Ok(addresses) => addresses,
+        Err(e) => {
+            return vec![Err((coin.ticker.clone(), e.into()))];
+        },
+    };
```

Comment:
When any of the account fails during the derivation, balance streaming on ETH will stop for all the addresses.

Reply:
`my_addresses` will never fail. Error handling is added because of the reliance on `derive_child` from `bip32` crate here https://github.com/KomodoPlatform/komodo-defi-framework/blob/ffd0c965d8d071e5f9d0e9f08cb5aa1847d3ac71/mm2src/coins/hd_wallet/coin_ops.rs#L63-L66 but we never pass to it any values that will cause errors.

Reply:
I think we should add an error log saying about it (like this is an unexpected error or something).